unit_tests/src/test_connection_pool.cpp

```diff
@@ -261,4 +262,100 @@ TEST_CASE("connection pool")
             REQUIRE(e.code() == SYS_SOCK_CONNECT_ERR);
         }
     }
+
+    SECTION("connection_pool_options")
+    {
+        irods::connection_pool_options cp_opts{};
+
+        SECTION("refresh connection after N retrievals")
```

Comment:
Should we add a test where a number of API requests are made on the same connection to make sure it is refreshed (or not, whatever the case may be)? I don't see where this is being implemented, but all of the comments/docs suggest that this is what `refresh_connections_after_n_retrievals` does, so I just want to make sure we're covering our bases.

Reply:
directly into a comment please :P

Reply:
Sure.

Reply:
Done

Reply:
Right-o. This was me trying to reconcile the docs with the implementation that I was seeing. Seems this is cleared up now